ethcore/src/engines/validator_set/contract.rs

```diff
@@ -0,0 +1,220 @@
+// Copyright 2015, 2016 Parity Technologies (UK) Ltd.
+// This file is part of Parity.
+
+// Parity is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// Parity is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with Parity.  If not, see <http://www.gnu.org/licenses/>.
+
+/// Validator set maintained in a contract.
+
+use std::sync::Weak;
+use util::*;
+use client::{Client, BlockChainClient};
+use client::chain_notify::ChainNotify;
+use super::ValidatorSet;
+use super::simple_list::SimpleList;
+
+/// The validator contract should have the following interface:
+/// [{"constant":true,"inputs":[],"name":"get_validators","outputs":[{"name":"","type":"address[]"}],"payable":false,"type":"function"}]
```

Comment:
Solidity functions are camelCase by convention (including in all our Soldity contracts).